pagefaultgames/pokerogue
# [Localization] Use gender context for run-history

## What are the changes the user will see?

Effectively none

## Why am I making these changes?

We should be using i18next features instead of custom development. 
I've migrated some keys & code in run-history to use [i18n context](https://www.i18next.com/translation-function/context) for gender specifics.

## What are the changes from a developer perspective?

For keys inside `run-history.json` that ended with `M` I've just removed the `M`.
For keys that ended with `F` I've replaced it with `_female`.

In the UI code I've replaced the gender-suffix logic with using `{ context: genderStr }` instead.

<sub>This change might also create new translation "voids" in [GitLocalize](https://gitlocalize.com/repo/9839). That's unfortunately a necessary evil</sub>

### Screenshots/Videos
<img width="933" alt="image" src="https://github.com/user-attachments/assets/fea0469f-2bb3-493b-a31f-dd2da314682e">

## How to test the changes?
- Start the game in a language that has gender specific run history keys
    - Import this run-history if necessary (replace `.txt` with `.prsv`) [runHistoryData_Guest.txt](https://github.com/user-attachments/files/16789861/runHistoryData_Guest.txt)
- Check the history
- Change your gender to a different one
- Check them again => Should render the gender context string

## Checklist
